src/main/java/de/tum/in/www1/exerciseapp/web/rest/FileUploadResource.java

```diff
@@ -0,0 +1,192 @@
+package de.tum.in.www1.exerciseapp.web.rest;
+
+import com.codahale.metrics.annotation.Timed;
+import de.tum.in.www1.exerciseapp.config.Constants;
+import de.tum.in.www1.exerciseapp.domain.DragAndDropQuestion;
+import de.tum.in.www1.exerciseapp.domain.DragItem;
+import de.tum.in.www1.exerciseapp.repository.DragAndDropQuestionRepository;
+import de.tum.in.www1.exerciseapp.repository.DragItemRepository;
+import de.tum.in.www1.exerciseapp.service.AuthorizationCheckService;
+import org.apache.commons.io.FilenameUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.time.ZonedDateTime;
+import java.util.UUID;
+
+/**
+ * REST controller for managing Course.
+ */
+@RestController
+@RequestMapping("/api")
+public class FileUploadResource {
+
+    private final AuthorizationCheckService authCheckService;
+    private final DragAndDropQuestionRepository dragAndDropQuestionRepository;
+    private final DragItemRepository dragItemRepository;
+
+    public FileUploadResource(AuthorizationCheckService authCheckService, DragAndDropQuestionRepository dragAndDropQuestionRepository, DragItemRepository dragItemRepository) {
+        this.authCheckService = authCheckService;
+        this.dragAndDropQuestionRepository = dragAndDropQuestionRepository;
+        this.dragItemRepository = dragItemRepository;
+    }
+
+    private final Logger log = LoggerFactory.getLogger(FileUploadResource.class);
+
+    /**
+     * POST  /fileUpload : Upload a new file.
+     *
+     * @param file The file to save
+     * @return The path of the file
+     */
+    @PostMapping("/fileUpload")
+    @PreAuthorize("hasAnyRole('ADMIN', 'INSTRUCTOR', 'TA')")
+    @Timed
+    public ResponseEntity<String> saveUserDataAndFile(@RequestParam(value = "file") MultipartFile file) throws URISyntaxException {
+        log.debug("REST request to upload file : {}", file.getOriginalFilename());
+
+        // check for size of file
+        if (file.getSize() > 5000000) {
```

Comment:
Do you know if the file is already loaded into memory here? If yes we should reject it earlier? Maybe as part of the http protocol?

Please also change it to 5 * 1024 * 1024 and save this as constant (preferably in the application-XXX.yml)

Could we additionally check the file size in the browser on the client side (preferably using the same constant from application-XXX.yml) and prevent it even being sent to the server?